app/validation/validator.py

```diff
@@ -120,82 +119,45 @@ def validate_answer(self, block, answer):
                 answer_errors.append(self._error_message(unrouted_error))
         return answer_errors
 
-    def validate_range_types_from_answers(self, json_to_validate):
+    def validate_numeric_answer_types(self, json_to_validate):
         errors = []
+        answer_ranges = {}
 
-        for block in self._get_blocks(json_to_validate):
-            for answer in self._get_answers_for_block(block):
-                used_answers = []
-                values = []
-                answer_id = answer['id']
-                answer_decimals = answer.get('decimal_places', 0)
+        numeric_answers = (
+            answer for answer in self._get_answers(json_to_validate)
+            if answer['type'] in ['Number', 'Currency', 'Percentage']
+        )
 
-                if answer.get('max_value') and 'value' in answer.get('max_value'):
-                    values.append(answer['max_value']['value'])
+        for answer in numeric_answers:
+            answer_ranges[answer.get('id')] = self._get_numeric_range_values(answer, answer_ranges)
 
-                if answer.get('max_value') and 'answer_id' in answer.get('max_value'):
-                    used_answers.append(answer['max_value']['answer_id'])
+            # Validate referred numeric answer exists (skip further tests for answer if error is returned)
+            if self._validate_referred_numeric_answer(answer, answer_ranges, errors):
+                continue
 
-                if answer.get('min_value') and 'value' in answer.get('min_value'):
-                    values.append(answer['min_value']['value'])
+            # Validate numeric answer has a positive range of possible responses
+            self._validate_numeric_range(answer, answer_ranges, errors)
```

Comment:
We should return a list of errors from this function and extend the errors list.
Modifying a function parameter is bad practice and its not clear that that parameter will be updated